metrics/storage/storage.go

```diff
@@ -89,6 +89,21 @@ func (ctx GCSDatastoreContext) Output(id OutputId, metadata interface{},
 	log.Printf("Writing %s to Google Cloud Storage\n", name)
 	gcsData := GCSData{metadata, data}
 	obj := ctx.Bucket.Handle.Object(id.DataLocation.GCSObjectPath)
+	objWriter := obj.NewWriter(ctx.Context)
+	if err := func() error {
+		gzWriter := gzip.NewWriter(objWriter)
+		encoder := json.NewEncoder(gzWriter)
+		if err := encoder.Encode(gcsData); err != nil {
+			return err
```

Comment:
If this errors, the writers won't close. The `defer` statements from the previous file-state should be kept.

If your intent is to have the write occur immediately, use flush:
https://golang.org/pkg/compress/gzip/#Writer.Flush

Reply:
Hmm. We are also trying to return potential error from closing `objWriter`. PTAL at my change: I defer closing `gzWriter` and manually close `objWriter` (discarding potential error) if we encounter an encoding error.